src/standalone/api/api.ts

```diff
@@ -133,10 +129,23 @@ export function buildApi(
             const dataViewerProviderService = serviceContainer.get<IDataViewerFactory>(IDataViewerFactory);
             await dataViewerProviderService.create(dataProvider, title);
         },
-        registerRemoteServerProvider(picker: IJupyterUriProvider): void {
+        registerRemoteServerProvider(provider: IJupyterUriProvider): IDisposable {
             sendApiUsageTelemetry(extensions, 'registerRemoteServerProvider');
             const container = serviceContainer.get<IJupyterUriProviderRegistration>(IJupyterUriProviderRegistration);
-            container.registerProvider(picker);
+            let disposeHook = noop;
+            const register = async () => {
+                const extensions = serviceContainer.get<IExtensions>(IExtensions);
+                const extensionId = provider.id.startsWith('_builtin')
+                    ? JVSC_EXTENSION_ID
+                    : (await extensions.determineExtensionFromCallStack()).extensionId;
+                container.registerProvider(Object.assign(provider, { extensionId }));
```

Comment:
I'm passing the extensionId from the very top and ensuring it is a property of the provider itself, 
this way, internally in our classes we know how owns a provider, else we need to work around this.
This is required for part of the larger refactoring to make it easier to determine the owners of each providcer

You might recall, i wanted to replace ServerId with `{extensionId; providerId; handle}`